Bad error message when missing pulumi-analyzer-policy executable

### What happened?

I use Manjaro (basically Arch), this is not the first time that the pulumi package from the Arch repo is missing some files.
The issue is that instead of printing something like this:
```
Could not find pulumi-analyzer-policy, it seems like your installation of pulumi is incomplete, check https://www.pulumi.com/docs/get-started/install/
```
It does this:
```console
$ pulumi preview --policy-pack ../crossguard-policies
Previewing update (xxxxxxx):
panic: fatal: An assertion has failed: failed to convert err no analyzer plugin 'pulumi-analyzer-policy' found in the workspace or on your $PATH, install the plugin using `pulumi plugin install analyzer policy` to Error, did this come from an RPC endpoint?

goroutine 135 [running]:
github.com/pulumi/pulumi/sdk/v3/go/common/util/contract.failfast(...)
        github.com/pulumi/pulumi/sdk/v3@v3.34.1/go/common/util/contract/f
[...]
```
And:
```console
$ pulumi plugin install analyzer policy
error: 404 HTTP error fetching plugin from https://api.github.com/repos/pulumi/pulumi-policy/releases/latest. If this is a private GitHub repository, try providing a token via the GITHUB_TOKEN environment variable. See: https://github.com/settings/tokens
```

### Steps to reproduce

```
$ docker run -it --rm archlinux bash
# pacman -Sy pulumi nodejs npm
# mkdir test && cd test
# pulumi login -l && pulumi new kubernetes-typescript
# mkdir p && echo "runtime: nodejs" > p/PulumiPolicy.yaml
# pulumi preview --policy-pack p/
```

=> optional step `rm -f /usr/bin/pulumi-analyzer-policy`, should not be needed as pulumi-analyzer-policy is missing.

### Expected Behavior

Print a friendly error message

### Actual Behavior

print a wall of text and errors, mislead the user

### Versions used

pulumi 3.34.1 from the Arch repo.

### Additional context

EDIT:
This is not the first time that this happens, but last time the error message that pulumi printed wasn't as bad https://bugs.archlinux.org/task/72394

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

Can repro the panic.

```
Previewing update (dev):
panic: fatal: An assertion has failed: failed to convert err no analyzer plugin 'pulumi-analyzer-policy' found in the workspace or on your $PATH to Error, did this come from an RPC endpoint?

    goroutine 43 [running]:
                           github.com/pulumi/pulumi/sdk/v3/go/common/util/contract.failfast(...)
   	github.com/pulumi/pulumi/sdk/v3@v3.78.0/go/common/util/contract/failfast.go:23
                                                                                      github.com/pulumi/pulumi/sdk/v3/go/common/util/contract.Assertf(0x0?, {0x5606d2ee7092?, 0x0?}, {0xc001709ae8?, 0x5606d3b1b490?, 0x10?})
                                   	github.com/pulumi/pulumi/sdk/v3@v3.78.0/go/common/util/contract/assert.go:35 +0xe8
                             github.com/pulumi/pulumi/sdk/v3/go/common/util/rpcutil/rpcerror.Convert({0x5606d3b0bb00?, 0xc00147b6b0})
                                        	github.com/pulumi/pulumi/sdk/v3@v3.78.0/go/common/util/rpcutil/rpcerror/rpcerror.go:196 +0x74
                                                github.com/pulumi/pulumi/sdk/v3/go/common/resource/plugin.NewPolicyAnalyzer({0x5606d3b3b768, 0xc00141efc0}, 0xc001442b40, {0xc00140fc30, 0x7}, {0xc00136add0, 0x2}, 0xc001431140?)
                                        	github.com/pulumi/pulumi/sdk/v3@v3.78.0/go/common/resource/plugin/analyzer_plugin.go:107 +0x385
                                                  github.com/pulumi/pulumi/sdk/v3/go/common/resource/plugin.(*defaultHost).PolicyAnalyzer.func1()
                                                    	github.com/pulumi/pulumi/sdk/v3@v3.78.0/go/common/resource/plugin/host.go:328 +0x153
                                               github.com/pulumi/pulumi/sdk/v3/go/common/resource/plugin.(*defaultHost).loadPlugin.func1()
                                             	github.com/pulumi/pulumi/sdk/v3@v3.78.0/go/common/resource/plugin/host.go:282 +0x22
                                      github.com/pulumi/pulumi/sdk/v3/go/common/resource/plugin.NewDefaultHost.func1()
                         	github.com/pulumi/pulumi/sdk/v3@v3.78.0/go/common/resource/plugin/host.go:151 +0x3d
                      created by github.com/pulumi/pulumi/sdk/v3/go/common/resource/plugin.NewDefaultHost in goroutine 1
                           	github.com/pulumi/pulumi/sdk/v3@v3.78.0/go/common/resource/plugin/host.go:149 +0x305
```

--- a/changelog/pending/20230824--engine--fix-panic-when-analyzer-plugin-is-not-found-on-path.yaml
+++ b/changelog/pending/20230824--engine--fix-panic-when-analyzer-plugin-is-not-found-on-path.yaml
@@ -1,0 +1,4 @@
+changes:
+- type: fix
+  scope: engine
+  description: Fix panic when analyzer plugin is not found on PATH.

--- a/sdk/go/common/resource/plugin/analyzer_plugin.go
+++ b/sdk/go/common/resource/plugin/analyzer_plugin.go
@@ -103,13 +103,15 @@ func NewPolicyAnalyzer(
 	// Load the policy-booting analyzer plugin (i.e., `pulumi-analyzer-${policyAnalyzerName}`).
 	pluginPath, err := workspace.GetPluginPath(ctx.Diag,
 		workspace.AnalyzerPlugin, policyAnalyzerName, nil, host.GetProjectPlugins())
-	if err != nil {
-		return nil, rpcerror.Convert(err)
-	} else if pluginPath == "" {
+
+	var e *workspace.MissingError
+	if errors.As(err, &e) {
 		return nil, fmt.Errorf("could not start policy pack %q because the built-in analyzer "+
 			"plugin that runs policy plugins is missing. This might occur when the plugin "+
 			"directory is not on your $PATH, or when the installed version of the Pulumi SDK "+
 			"does not support resource policies", string(name))
+	} else if err != nil {
+		return nil, err
 	}
 
 	// Create the environment variables from the options.

